pure.zsh

```diff
@@ -222,6 +222,13 @@ prompt_pure_precmd() {
 		export VIRTUAL_ENV_DISABLE_PROMPT=12
 	fi
 
+	# Nix package manager integration. If used from within 'nix shell' - shell name is shown like so:
+	# ~/Projects/flake-utils-plus master
+	# flake-utils-plus ❯
+	if [[ -n $IN_NIX_SHELL ]]; then
+		psvar[12]="${name:-nix-shell}"
```

Comment:
Should we consider compatibility with e.g. virtualenvs? Can there be a virtualenv active simultaneously with a nix shell? Should both be visible?

Reply:
Yeah, I thought about it and decided to implement it in this manner since showing both venv and nix-shell would not make much sense.

And since nix-shell may contain venv and not another way around - it is implemented in a manner so it would overwrite venv name